servo/repeating.py

```diff
@@ -62,13 +71,28 @@ def start_repeating_task(
 
         async def repeating_async_fn() -> None:
             while True:
-                callable()
-                await asyncio.sleep(every.total_seconds())
+                t0 = time.time()
+                if asyncio.iscoroutinefunction(function):
+                    await function()
+                elif callable(function):
+                    function()
+                else:
+                    raise TypeError(f"function={function} must be Awaitable or Callable, but has "
+                        f"type(function)={type(function)}.")
+                t1 = time.time()
+                sleep_time = every.total_seconds()
+                if time_correction:
+                    sleep_time -= max(t1 - t0, 0)
+                if sleep_time > 0:
+                    await asyncio.sleep(sleep_time)
+                else:
+                    await asyncio.sleep(float_info.epsilon)
 
         asyncio_task = asyncio.create_task(repeating_async_fn(), name=task_name)
         self._repeating_tasks[name] = asyncio_task
         return asyncio_task
 
+
```

Comment:
Remove this for consistency of whitespacing with other functions in this file

Reply:
We need a `black` (or other) formatting check in the CI to resolve these kinds of things.

Reply:
That sounds awesome, LMK what you have in mind